### PR TITLE
Add outsourced auth post

### PR DESCRIPTION
**Pull request by mooreds**

### CODE REVIEW

File: site/_posts/2021-01-13-why-outsource-auth.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+excerpt_separator: "<!--more-->"
+---
+
+You're a software engineering leader, and you're great at your job. You know that the optimal path for software development lies in figuring out which components of your design to implement from scratch and which have already been implemented by specialists and can be reused. 
+
+<!--more-->
+
+You also know that these aren't decisions that you can only make once -- you have to keep reevaluating based on environment changes and the needs of new products.
+
+Authentication is one of those components that you deal with all the time. Auth is a necessary part of any web-based software product, but how you implement auth is not necessarily always the same. Careful consideration is needed, because your decision to outsource will not only impact speed of development, but also long-term product maintenance -- you don't want to slow down time to market because you re-implemented an entire auth system unnecessarily, but you also don't want to use an auth system that is going to cause problems down the road.
```
Comment (voidmain) [inline site/_posts/2021-01-13-why-outsource-auth.md:17]: I wouldn't limit this to web-based apps or products.

File: site/_posts/2021-01-13-why-outsource-auth.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+Properly storing passwords is an incredibly low bar, and yet companies that manage their own auth still do it incorrectly all the time.
+
+## Consequences of an auth outage
+
+While less damaging than breaches, outages can still cause reputational damage and liability issues if your SLAs make uptime guarantees. Similarly to breaches, if you outsource your auth system it's likely that any auth outage will extend beyond your company. 
+
+As an example, when Microsoft's Azure Active Directory (AAD) [went down for a good portion of the afternoon late last year](https://www.zdnet.com/article/microsofts-azure-ad-authentication-outage-what-went-wrong/), logins for applications across the internet stopped working. 
+
+When your competitors' authorization systems are down at the same time yours are, nobody blames you for it, but when you're the only company having issues, you suffer reputationally. No matter what your outsourced auth system is (FusionAuth, Cognito, AAD, etc), you can be almost certain that you won't be alone in the event of an outage.
+
```
Comment (voidmain) [inline site/_posts/2021-01-13-why-outsource-auth.md:46]: I think the Pokemon Go example is great here if we can add it. They built their own auth and it basically caused millions of players to be unable to even try the game when it first came out. They scrambled to patch it and that did very little. It definitely had a major impact on their brand and reputation.

File: site/_posts/2021-01-13-why-outsource-auth.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+However, the drawbacks can be large:
+
+* Complex new features can take significant time to build. Outsourced auth systems likely have these features already built (things like multi-factor authentication, user management interfaces, analytics and audit logs, and brute force hack detection, among others).
+* With an in-house solution, you'll have to budget time to monitor new security threats and patch your system in a rapidly evolving threat landscape.
+* You won't get the benefits of a dedicated team that are constantly improving your auth system. This is actually a bigger deal than it seems, because if you outsource auth then other companies will also be filing issue reports and feature requests on your behalf, so you reap the benefits of those extra eyes as well.
+* In the case of mergers or acquisitions, an in-house solution is likely to be terrible at combining different databases of users and managing things like duplicates and incomplete data. Enterprise Identity Unification (EIU) efforts can founder on internal auth systems.
```
Comment (voidmain) [inline site/_posts/2021-01-13-why-outsource-auth.md:59]: We might need a link to something that describes EIU here. I think the last sentence isn't correct either. Should it be "flounder"? This entire bullet could use some work.

File: site/_posts/2021-01-13-why-outsource-auth.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+## Cost
+
+When building an in-house auth system your costs are all ostensibly sunk (engineer salaries). However, if building your system in-house delays time to market or prevents creation of other features, the build could cost you a significant amount of real income. There will also be on-going maintenance costs with an in-house solution. 
+
+When making cost calculations, you should compare:
+
+**The revenue lost due to getting to market later PLUS the cost of engineer time to implement PLUS on-going staff hours required for maintenance**
```
Comment (voidmain) [inline site/_posts/2021-01-13-why-outsource-auth.md:67]: Should we make these real equations? If so, then the first one is more complex:

"Revenue lost by slower time-to-market" + "engineering cost to implement in-house solution" + "on-going maintenance costs of in-house solution" + "increased risk of data breach" + "increased risk of outage" + "increased risk during a merger or acquisition"

File: site/_posts/2021-01-20-why-outsource-auth.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+However, the drawbacks can be large:
+
+* Complex new features can take significant time to build. Outsourced auth systems likely have these features already built (things like multi-factor authentication, user management interfaces, analytics and audit logs, and brute force hack detection, among others).
+* With an in-house solution, you'll have to budget time to monitor new security threats and patch your system in a rapidly evolving threat landscape.
+* You won't get the benefits of a dedicated team that are constantly improving your auth system. This is actually a bigger deal than it seems, because if you outsource auth then other companies will also be filing issue reports and feature requests on your behalf, so you reap the benefits of those extra eyes as well.
+* In the case of mergers or acquisitions, an in-house solution is likely to be terrible at combining different databases of users and managing things like duplicates or incomplete data. Such enterprise identity unification efforts can founder on internal auth systems. FusionAuth, on the other hand, supports modeling different user bases with tenants. 
+
+## Cost
```
Comment (dpolstra) [inline site/_posts/2021-01-20-why-outsource-auth.md:61]: cost of in-house vs outsourced auth

**Pull request merged**